src/xrpld/app/tx/detail/VaultWithdraw.cpp

```diff
@@ -313,6 +320,64 @@ VaultWithdraw::doApply()
         return account_;
     }();
 
+    auto const reserveCheck = [&, this](AccountID dstAcct, TER error) -> TER {
+        // Enforced by `requireAuth` called in `preclaim`
+        if (dstAcct != account_)
+            return tecINTERNAL;  // LCOV_EXCL_LINE
+
+        auto const sleAcct = view().read(keylet::account(account_));
+        if (!sleAcct)
+            return tecINTERNAL;  // LCOV_EXCL_LINE
+
+        // Can the account cover the MPToken or trust line reserve ?
+        std::uint32_t const ownerCount = sleAcct->at(sfOwnerCount);
+        if (mPriorBalance < view().fees().accountReserve(ownerCount + 1))
+            return error;
+
+        return tesSUCCESS;
+    };
+
+    if (!vaultAsset.native() && dstAcct != vaultAsset.getIssuer())
+    {
+        if (auto const ter = std::visit(
+                [&]<ValidIssueType TIss>(TIss const& issue) -> TER {
+                    if constexpr (std::is_same_v<TIss, Issue>)
+                    {
+                        if (auto const sleLine =
+                                view().read(keylet::line(dstAcct, issue));
+                            sleLine == nullptr)
+                            return reserveCheck(
+                                dstAcct, tecNO_LINE_INSUF_RESERVE);
+
+                        // If the above succeeded then the required trust line
+                        // will be created indirectly by `accountSend` below.
+                    }
+                    else
+                    {
+                        if (auto const sleMPT = view().read(
+                                keylet::mptoken(issue.getMptID(), account_));
+                            sleMPT == nullptr)
+                        {
+                            if (auto const ter = reserveCheck(
+                                    dstAcct, tecINSUFFICIENT_RESERVE);
+                                !isTesSuccess(ter))
+                                return ter;
+
+                            if (auto const ter =
+                                    MPTokenAuthorize::createMPToken(
+                                        view(), issue.getMptID(), account_, 0);
+                                !isTesSuccess(ter))
+                                return ter;  // LCOV_EXCL_LINE
```

Comment:
the only possible error here is `tecDIR_FULL` which we normally do not test for.